File: backend/src/main/java/org/camunda/optimize/service/security/Session.java
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under one or more contributor license agreements.
 * Licensed under a proprietary license. See the License.txt file for more information.
 * You may not use this file except in compliance with the proprietary license.
 */
package org.camunda.optimize.service.security;

public class Session {

  private TokenVerifier tokenVerifier;

  Session(TokenVerifier tokenVerifier) {
    this.tokenVerifier = tokenVerifier;
  }

  public void updateExpiryDate() {
    tokenVerifier.updateExpiryDate();
  }

  public boolean isTokenValid(String token) {
    return tokenVerifier.isTokenValid(token);
  }

  public boolean hasTokenExpired(String token) {
    return tokenVerifier.hasExpired(token);
  }

}

File: operate/client/e2e-playwright/utils/session.ts
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import fs from 'node:fs';

const sessionFile = 'playwright/.auth/session.json';

const readSessionStorage = (): Record<string, string> => {
  if (!fs.existsSync(sessionFile)) {
    return {};
  }

  return JSON.parse(fs.readFileSync(sessionFile, 'utf-8')) as Record<
    string,
    string
  >;
};

const getCsrfToken = (): string | undefined => {
  return readSessionStorage()['X-CSRF-TOKEN'];
};

export {sessionFile, readSessionStorage, getCsrfToken};
